q2_fmt/tests/test_engraftment.py

```diff
@@ -487,3 +488,364 @@ def test_alpha_dists_with_extra_samples_in_diversity_not_in_metadata(self):
 
     def test_examples(self):
         self.execute_examples()
+
+
+class TestPeds(TestBase):
+    def test_check_get_donor(self):
+        metadata_df = pd.DataFrame({
+            'id': ['sample1', 'sample2', 'sample3', 'sample4',
+                   'donor1', 'donor2'],
+            'Ref': ['donor1', 'donor1', 'donor2', 'donor2', float("Nan"),
+                    float("Nan")],
+            'subject': ['sub1', 'sub1', 'sub2', 'sub2', float("Nan"),
+                        float("Nan")],
+            'group': [1, 2, 1, 2, float("Nan"),
+                      float("Nan")]}).set_index('id')
+        print(metadata_df)
+        reference_series = metadata_df['Ref'].dropna()
+        table_df = pd.DataFrame({
+            'id': ['sample1', 'sample2', 'sample3', 'sample4',
+                   'donor1', 'donor2'],
+            'Feature1': [1, 0, 1, 1, 1, 1],
+            'Feature2': [1, 1, 1, 1, 1, 1],
+            'Feature3': [0, 0, 1, 1, 1, 1]}).set_index('id')
+        peds_df = _compute_peds(reference_series=reference_series,
+                                table=table_df, metadata=metadata_df,
+                                time_column="group",
+                                reference_column="Ref",
+                                subject_column="subject")
+        peds_df = peds_df.set_index("id")
+        donor = peds_df.at["sample1", "donor"]
+        self.assertEqual(donor, "donor1")
+
+    def test_check_get_subject(self):
```

Comment:
```suggestion
    def test_get_subject(self):
```
Same suggestion as `test_get_donor`